sdk/python/kfp/v2/compiler/compiler_test.py

```diff
@@ -323,6 +324,74 @@ def my_pipeline(input1):
       compiler.Compiler().compile(
           pipeline_func=my_pipeline, package_path='output.json')
 
+  def test_passing_generic_artifact_to_input_expecting_concrete_artifact(
+      self):
+
+    producer_op1 = components.load_component_from_text("""
+      name: producer compoent
+      outputs:
+      - {name: output, type: Artifact}
+      implementation:
+        container:
+          image: dummy
+          args:
+          - {outputPath: output}
+      """)
+
+    @dsl.component
+    def producer_op2(output: dsl.Output[dsl.Artifact]):
+      pass
+
+    @dsl.component
+    def consumer_op(input: dsl.Input[dsl.Dataset]):
```

Comment:
Nice. Can we also add an example where consumption is using component.yaml with some random string as the artifact type (e.g. `MyModel` or something)?

Reply:
Done.